Sources/UI/Device/Device.swift

```diff
@@ -0,0 +1,405 @@
+        var systemInfo = utsname()
+        uname(&systemInfo)
+        let size = MemoryLayout<CChar>.size
+        let modelCode = withUnsafePointer(to: &systemInfo.machine) {
+            $0.withMemoryRebound(to: CChar.self, capacity: size) {
+                String(cString: UnsafePointer<CChar>($0))
+            }
+        }
+        
+        guard let modelName = String(validatingUTF8: modelCode) else {
+            fatalError("Invalid data")
+        }
+        
+        guard let deviceModel = DeviceModel(modelName: modelName) else {
+            os_log("Unknown model name: %@", log: .context, type: .error, modelName)
+            return modelName
+        }
+        
+        return deviceModel.description
+    }
+    
+    public var deviceName: String {
+        return UIDevice.current.name
+    }
+    
+    public var operatingSystemName: String {
+        return UIDevice.current.systemName
+    }
+    
+    public var operatingSystemVersion: String {
+        return UIDevice.current.systemVersion
+    }
+    
+    public var screenHeight: Int {
+        return Int(UIScreen.main.bounds.height)
+    }
+    
+    public var screenWidth: Int {
+        return Int(UIScreen.main.bounds.width)
+    }
+    
+    public var sdkVersion: String {
+        let bundle: Bundle = {
+            if let bundle = Bundle(identifier: "io.rover.RoverFoundation") {
+                return bundle
+            }
+            
+            if let bundle = Bundle(identifier: "org.cocoapods.RoverKit") {
+                return bundle
+            }
+            
+            fatalError("No bundle found with identifier io.rover.RoverFoundation or org.cocoapods.RoverKit")
+        }()
+        
+        return bundle.infoDictionary!["CFBundleShortVersionString"] as! String
+    }
+    
+    // MARK: Time Zone
+    
+    public var timeZone: String {
+        return (NSTimeZone.local as NSTimeZone).name
+    }
+    
+    // MARK: User Info
+    
+    public private(set) var userInfo: Attributes? {
+        get {
+            guard let data = userDefaults.data(forKey: "io.rover.RoverData.userInfo") else {
+                return nil
+            }
+            
+            do {
+                return try jsonDecoder.decode(Attributes.self, from: data)
+            } catch {
+                os_log("Failed to decode user info: %@", log: .general, type: .error, error.localizedDescription)
+                return nil
+            }
+        }
+        set {
+            do {
+                let data = try jsonEncoder.encode(newValue)
+                userDefaults.set(data, forKey: "io.rover.RoverData.userInfo")
+            } catch {
+                os_log("Failed to encode user info: %@", log: .general, type: .error, error.localizedDescription)
+            }
+        }
+    }
+    
+    // MARK: Notifications Authorization
+    
+    public var notificationAuthorization: String {
+        // Refresh status for _next_ time context is requested
```

Comment:
Update comment since we're moving away from "context". Suggested re-wording: "Refresh status next time snapshot is created".